projects/packages/sync/src/modules/class-term-relationships.php

```diff
@@ -119,10 +119,10 @@ public function enqueue_full_sync_actions( $config, $max_items_to_enqueue, $last
 			 */
 			$objects = $wpdb->get_results( $wpdb->prepare( "SELECT object_id, term_taxonomy_id FROM $wpdb->term_relationships WHERE ( object_id = %d AND term_taxonomy_id < %d ) OR ( object_id < %d ) ORDER BY object_id DESC, term_taxonomy_id DESC LIMIT %d", $last_object_enqueued['object_id'], $last_object_enqueued['term_taxonomy_id'], $last_object_enqueued['object_id'], $limit ), ARRAY_A );
 			// Request term relationships in groups of N for efficiency.
-			$objects_count = is_countable( $objects ) ? count( $objects ) : 0;
-			if ( ! $objects_count ) {
+			if ( ! is_countable( $objects ) || count( $objects ) === 0 ) {
 				return array( $items_enqueued_count, true );
 			}
+			$objects_count         = count( $objects );
```

Comment:
I'm not sure I understand the purpose of this change. If it's just to improve readability, maybe this is better?

```
$objects_count = is_countable( $objects ) ? count( $objects ) : 0;
if ( $objects_count === 0 ) {
    return array( $items_enqueued_count, true );
}
```

Reply:
Ah, I see in the changelog it's probably some Phan peculiarity.

Reply:
Phan is smart enough to realize that
```
if ( ! is_countable( $objects ) ) {
    return;
}
// $objects is definitely countable at this point
```
But it's not smart enough to reason that through the assignment of 0 to `$objects_count` and then the test for `$objects_count` being 0 or falsey. Having it be able to know that it's countable saves it from complaining about it being possibly null or something else a few lines down.